src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs

```diff
@@ -173,4 +174,421 @@ private static bool IsTls13Disabled(bool isServer)
 #endif
         return false;
     }
+
+    public void SetContext(MsQuicSafeHandle handle, void* context)
+    {
+        bool success = false;
+        try
+        {
+            handle.DangerousAddRef(ref success);
+            Debug.Assert(success || handle.IsInvalid);
+            ObjectDisposedException.ThrowIf(handle.IsInvalid, handle);
+            ApiTable->SetContext(handle.QuicHandle, context);
+        }
+        finally
+        {
+            if (success)
+            {
+                handle.DangerousRelease();
+            }
+        }
+    }
+
+    public void* GetContext(MsQuicSafeHandle handle)
+    {
+        bool success = false;
+        try
+        {
+            handle.DangerousAddRef(ref success);
+            Debug.Assert(success || handle.IsInvalid);
+            ObjectDisposedException.ThrowIf(handle.IsInvalid, handle);
+            return ApiTable->GetContext(handle.QuicHandle);
+        }
+        finally
+        {
+            if (success)
+            {
+                handle.DangerousRelease();
+            }
+        }
+    }
+
+    public void SetCallbackHandler(MsQuicSafeHandle handle, void* callback, void* context)
+    {
+        bool success = false;
+        try
+        {
+            handle.DangerousAddRef(ref success);
+            Debug.Assert(success || handle.IsInvalid);
+            ObjectDisposedException.ThrowIf(handle.IsInvalid, handle);
```

Comment:
This should not be needed. `handle.DangerousAddRef` will throw `ObjectDisposedException` if the safe handle was disposed already.

Reply:
Thanks. I now see in the source code that if `DangerousAddRef` returned without throwing, `success` can never be false, so both this line and the previous one are excessive.

Reply:
Done.